runtime/autoload/man.vim

```diff
@@ -190,6 +190,23 @@ function! s:open_page(sect, name)
     silent keepjumps 1delete _
   endwhile
   setlocal filetype=man
+  call man#create_toc()
+endfunction
+
+function! man#create_toc() abort
+  if !exists('b:man_toc')
+    " Cache the TOC since user might switch in and out between the man pages.
+    let b:man_toc = []
+    for lnum in range(1, line('$'))
+      let text = getline(lnum)
+      if text =~# '^\%(\%(\S.*\)\=\S\|\%1l.*\|\s\{3\}\S.*\)$'
```

Comment:
I'm gonna update the syntax file to change the last regex there to instead of `\s`, just use ` `. So it looks like ` \{3\}\S.*`.